test/features/authentication/data/repositories/auth_repository_test.dart

```diff
@@ -0,0 +1,577 @@
+  MockSpec<UserCredential>(),
+  MockSpec<User>(),
+  MockSpec<Users>(),
+  MockSpec<File>(),
+  MockSpec<UploadTask>(),
+  MockSpec<SharedPreferences>(),
+])
+import 'auth_repository_test.mocks.dart';
+
+void main() {
+  late FirebaseAuth firebaseAuth;
+  late FirebaseFirestore firestore;
+  late FirebaseStorage firebaseStorage;
+  late SharedPreferences sharedPreferences;
+  late AuthRepository authRepository;
+
+  setUp(() {
+    firebaseAuth = MockFirebaseAuth();
+    firestore = MockFirebaseFirestore();
+    firebaseStorage = MockFirebaseStorage();
+    sharedPreferences = MockSharedPreferences();
+    authRepository = AuthRepository(
+        firestore, firebaseAuth, firebaseStorage, sharedPreferences);
+  });
+
+  group('logInWithEmailAndPassword', () {
+    late String email;
+    late String password;
+    late UserCredential userCredential;
+
+    setUp(() {
+      email = 'example@ex.com';
+      password = '123456';
+      userCredential = MockUserCredential();
+    });
+    test(
+        'logInWithEmailAndPassword calls signInWithEmailAndPassword with correct arguments',
+        () async {
+      when(
+        firebaseAuth.signInWithEmailAndPassword(
+          email: email,
+          password: password,
+        ),
+      ).thenAnswer((_) => Future<UserCredential>.value(userCredential));
+      await authRepository.logInWithEmailAndPassword(
+        email: email,
+        password: password,
+      );
+      verify(
+        firebaseAuth.signInWithEmailAndPassword(
+          email: email,
+          password: password,
+        ),
+      ).called(1);
+    });
+    test(
+        'logInWithEmailAndPassword throws an exception if signInWithEmailAndPassword throws',
+        () async {
+      when(firebaseAuth.signInWithEmailAndPassword(
+        email: email,
+        password: password,
+      )).thenThrow(FirebaseAuthException(code: 'invalid-email'));
+
+      expect(
+        () => authRepository.logInWithEmailAndPassword(
+            email: email, password: password),
+        throwsA(isA<FirebaseAuthException>()),
+      );
+    });
+  });
+
+  group('forgetEmailPassword', () {
+    late String email;
+
+    setUp(() {
+      email = 'example@ex.com';
+    });
+    test(
+        'forgetEmailPassword calls sendPasswordResetEmail with correct arguments',
+        () async {
+      await authRepository.forgetEmailPassword(email: email);
+      verify(
+        firebaseAuth.sendPasswordResetEmail(
+          email: email,
+        ),
+      ).called(1);
+    });
+    test(
+        'forgetEmailPassword throws an exception if sendPasswordResetEmail throws',
+        () async {
+      when(firebaseAuth.sendPasswordResetEmail(
+        email: email,
+      )).thenThrow(FirebaseAuthException(code: 'invalid-email'));
+
+      expect(
+        () => authRepository.forgetEmailPassword(email: email),
+        throwsA(isA<FirebaseAuthException>()),
+      );
+    });
+  });
+
+  // group('verifyPhoneNumber', () {
```

Comment:
@rfaturriza what would happen if test case group `verifyPhoneNumber` is enabled?

Reply:
@rfaturriza ignore my comment above, I just see the `verifyPhoneNumber` test group is enabled in the last commit

Reply:
It's outdated file. You can check the new commit here https://github.com/peltops/e-imunisasi/pull/100/commits/afdca06b4f2d47e83a319887478eb791ff4eb9fe